src/util/nondet.h

```diff
@@ -9,6 +9,7 @@ Author: Diffblue Ltd.
 #ifndef CPROVER_JAVA_BYTECODE_NONDET_H
 #define CPROVER_JAVA_BYTECODE_NONDET_H
 
+#include <util/allocate_objects.h>
 #include <util/std_code.h>
 #include <util/std_expr.h>
 #include <util/symbol_table.h>
```

Comment:
I guess it's the first time I'm properly looking at this: the only includes we really need here are `#include "std_code.h"` and `#include "std_expr.h"`. Everything else can (and IMHO: should be) done via forward declarations.

Reply:
Done in a new commit.